src/libawkward/Content.cpp

```diff
@@ -1189,6 +1189,37 @@ namespace awkward {
                                           index.length());
   }
 
+  const ContentPtr getitem_next_missing_jagged(const SliceMissing64& missing,
+                                               const Slice& tail,
+                                               const Index64& advanced,
+                                               const ContentPtr& that) {
+    const SliceJagged64* jagged = dynamic_cast<SliceJagged64*>(missing.content().get());
+    const Index64 index = missing.index();
+    ContentPtr content = that.get()->getitem_at_nowrap(0);
+    if ( content.get()->length() != index.length() ) {
+      throw std::invalid_argument("Contents length does not match length of masked jagged slice");
+    }
```

Comment:
The content length is allowed to be longer than the length that its parent can see, and hence longer than the slice, so I think the error condition should be `content.get()->length() < index.length() `.

Reply:
Here the parent is the "dummy" length-1 RegularArray wrapper. I think the content should exactly match the length of the mask, otherwise we allow things like:
```
a = awkward1.Array([[0, 1, 2], [3, 4], [5, 6], [7]])
b = awkward1.Array([[0], [0], [0], [0], [0], [0]])
a[b]
```
which I don't see a good reason to allow

Reply:
Contents are allowed to be longer than the nodes that contain them (i.e. have unreachable elements) because of how they're sometimes created: making several arrays is not an atomic operation, but if appends are performed from the innermost to the outermost, allowing long contents to be valid means that the whole array is valid for each of the atomic steps.

An example is if ArrayBuilder is filling some interior content with `begin_list` or `begin_record` that hasn't yet been paired with an `end_list` or `end_record`. During that time, the array as a whole is valid; it just doesn't have the partial list or record, though you might see some of the partial contents if you investigate its `content` during this time.

As for slices, I think the logic carries over. I think.

Reply:
I changed it as you suggested. Hopefully it doesn't surprise us later